src/data_processing/utils.py

```diff
@@ -558,3 +556,82 @@ def make_session() -> requests.Session:
     http.mount("https://", adapter)
     http.mount("http://", adapter)
     return http
+
+
+def are_identical_nc_files(filename1: str, filename2: str) -> bool:
+    with netCDF4.Dataset(filename1, "r") as nc1, netCDF4.Dataset(filename2, "r") as nc2:
+        try:
+            _compare_dimensions(nc1, nc2)
+            _compare_global_attributes(nc1, nc2)
+            _compare_variables(nc1, nc2)
+            _compare_variable_attributes(nc1, nc2)
+        except AssertionError as err:
+            logging.debug(err)
+            return False
+    return True
+
+
+def _compare_dimensions(nc1: netCDF4.Dataset, nc2: netCDF4.Dataset):
+    dims1 = nc1.dimensions.keys()
+    dims2 = nc2.dimensions.keys()
+    assert len(set(dims1) ^ set(dims2)) == 0, f"different dimensions: {dims1} vs {dims2}"
+    for dim in nc1.dimensions:
+        value1 = len(nc1.dimensions[dim])
+        value2 = len(nc2.dimensions[dim])
+        assert value1 == value2, _log("dimensions", dim, value1, value2)
+
+
+def _compare_global_attributes(nc1: netCDF4.Dataset, nc2: netCDF4.Dataset):
+    attribute_skips = (
+        "history",
+        "cloudnetpy_version",
+        "cloudnet_processing_version",
+        "file_uuid",
+    )
+    l1 = [a for a in nc1.ncattrs() if a not in attribute_skips]
+    l2 = [a for a in nc2.ncattrs() if a not in attribute_skips]
+    assert len(set(l1) ^ set(l2)) == 0, f"different global attributes: {l1} vs. {l2}"
+    for name in l1:
+        value1 = getattr(nc1, name)
+        value2 = getattr(nc2, name)
+        assert value1 == value2, _log("global attributes", name, value1, value2)
+
+
+def _compare_variables(nc1: netCDF4.Dataset, nc2: netCDF4.Dataset):
+    vars1 = nc1.variables.keys()
+    vars2 = nc2.variables.keys()
+    assert len(set(vars1) ^ set(vars2)) == 0, f"different variables: {vars1} vs. {vars2}"
+    for name in vars1:
+        value1 = nc1.variables[name][:]
+        value2 = nc2.variables[name][:]
+        assert value1.shape == value2.shape
+        assert ma.count_masked(value1) == ma.count_masked(value2)
+        if hasattr(value1, "mask") and hasattr(value2, "mask"):
+            value1 = value1[~value1.mask]
+            value2 = value2[~value2.mask]
+        assert np.isclose(value1, value2, rtol=1e-4).all(), _log(
+            "variable values", name, value1, value2
+        )
```

Comment:
Tämän pitäisi tehdä sama:
```suggestion
        assert ma.allclose(value1, value2, rtol=1e-4), _log(
            "variable values", name, value1, value2
        )
```

Reply:
Aah, hyvä löytö! :+1: